docs/source/guides/feature_descriptions.rst

```diff
@@ -0,0 +1,116 @@
+                          agg_primitives=["mean", "sum", "mode", "n_most_common"],
+                          trans_primitives=["month", "hour"],
+                          max_depth=2,
+                          features_only=True)
+
+By default, ``describe_feature`` uses the existing variable and entity names and the default primitive description templates to generate feature descriptions. 
+
+.. ipython:: python
+
+    feature_defs[7]
+    ft.describe_feature(feature_defs[7])
+
+.. ipython:: python
+
+    feature_defs[12]
+    ft.describe_feature(feature_defs[12])
+
+Improving Descriptions
+~~~~~~~~~~~~~~~~~~~~~~~
+While the default descriptions can be helpful, they can also be further improved by providing custom definitions of variables and features, and by providing alternative templates for primitive descriptions. 
+
+Feature Descriptions
+---------------------
+Custom feature definitions will get used in the description in place of the automatically generated description. This can be used to better explain what a variable or feature is, or to provide descriptions that take advantage of a user's existing knowledge about the data or domain. 
+
+.. ipython:: python
+
+    feature_descriptions = {
+        'customers: join_date': 'the date the customer joined'}
+
+    ft.describe_feature(feature_defs[8], feature_descriptions=feature_descriptions)
+
+For example, the above replaces the variable name ``"join_date"`` with a more descriptive definition of what that variable represents in the dataset. Feature descriptions can also be provided for generated features.
+
+.. ipython:: python
+
+    feature_descriptions = {
+        'sessions: SUM(transactions.amount)': 'the total transaction amount for a session'}
+
+    ft.describe_feature(feature_defs[12], feature_descriptions=feature_descriptions)
+
+
+Here, we create and pass in a custom description of the intermediate feature ``SUM(transactions.amount)``. The description for ``MEAN(sessions.SUM(transactions.amount))``, which is built on top of ``SUM(transactions.amount)``, uses the custom description in place of the automatically generated one. Feature descriptions can be passed in as a dictionary that maps the custom descriptions to either the feature object itself or the unique feature name in the form ``"[entity_name]: [feature_name]"``, as shown above.
+
+Primitive Templates
+--------------------
+Primitives descriptions are generated using primitive templates. By default, these are defined using the ``description_template`` attribute on the primitive. Primitives without a template default to using the ``name`` attribute of the primitive if it is defined, or the class name if it is not. Primitive description templates are string templates that take input feature descriptions as the positional arguments. These can be overwritten by mapping primitive instances or primitive names to custom templates and passing them into ``describe_feature`` through the ``primitive_templates`` argument. 
```

Comment:
Maybe we add a final sentence explaining the below example, mentioning that we're overriding the default "average of" template with our custom "total of" template